rkt/image_rm.go

```diff
@@ -39,34 +39,51 @@ func rmImages(s *store.Store, images []string) error {
 	errors := 0
 	staleErrors := 0
 	for _, pkey := range images {
+		var keys []string
 		errors++
 		h, err := types.NewHash(pkey)
 		if err != nil {
-			stderr("rkt: wrong image ID %q: %v", pkey, err)
-			continue
-		}
-		key, err := s.ResolveKey(h.String())
-		if err != nil {
-			stderr("rkt: image ID %q not valid: %v", pkey, err)
-			continue
-		}
-		if key == "" {
-			stderr("rkt: image ID %q doesn't exist", pkey)
-			continue
+			var found bool
+			keys, found, err = s.ResolveName(pkey)
+			if len(keys) > 0 {
+				errors += len(keys) - 1
+			}
+			if !found {
+				stderr("rkt: image name %q not found", pkey)
+				continue
+			}
+			if err != nil {
```

Comment:
You might never go in this `err != nil` branch because in case of error, `found` is most likely `false` so the previous `if` will skip this iteration of the loop.

I think you should test `err != nil` before testing `!found`.